src/main/java/Growler.java

```diff
@@ -0,0 +1,129 @@
+import java.util.Scanner;
+
+
+public class Growler {
+    private static Task[] tasks = new Task[100];
+    static int taskIndex = 0;
+    public static void main(String[] args) {
+        String greet = "Hello! I'm Growler \nWhat can I do for you? \n";
+        String Farewell = "Bye. Hope to see you again soon!";
+        System.out.println(greet);
+        String userInput;
+        Scanner in = new Scanner(System.in);
+        while (true) {
+            userInput = in.nextLine();
+            if (userInput.equals("bye")) {
+                System.out.println(Farewell);
+                break;
+            } else if (userInput.startsWith("mark")) {
+                // Validate input and mark the task as done
+                if (!isValidTaskCommand(userInput, "mark")) {
+                    System.out.println("Invalid format! Use: mark <task number>");
+                    return;
+                }
+                markList(userInput, true);
+            } else if (userInput.startsWith("unmark")) {
+                // Validate input and mark the task as not done
+                if (!isValidTaskCommand(userInput, "unmark")) {
+                    System.out.println("Invalid format! Use: unmark <task number>");
+                    return;
+                }
+                markList(userInput, false);
+            } else if (userInput.equals("list")) {
+                showList();
+            } else {
+                addTask(userInput);
+            }
+        }
+        in.close();
+    }
+    private static void addTask(String userInput) {
```

Comment:
This code is quite long, splitting it up into separate functions(per task) would be better